dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileHandler.java

```diff
@@ -0,0 +1,142 @@
+   * Attributes)}.
+   */
+  private ArrayDeque<SymbolProblemMatcher> stack = new ArrayDeque<>();
+
+  private ImmutableList.Builder<LinkageErrorMatcher> matchers;
+
+  ImmutableList<LinkageErrorMatcher> getMatchers() {
+    return matchers.build();
+  }
+
+  @Override
+  public void startDocument() {
+    if (matchers != null) {
+      throw new IllegalStateException("This handler started reading document already");
+    }
+    matchers = ImmutableList.builder();
+  }
+
+  private void addMatcherToTop(Object child) throws SAXException {
+    SymbolProblemMatcher parent = stack.peek();
+    if (parent instanceof SourceMatcher && child instanceof SymbolProblemSourceMatcher) {
+      ((SourceMatcher) parent).setMatcher((SymbolProblemSourceMatcher) child);
+    } else if (parent instanceof TargetMatcher && child instanceof SymbolProblemTargetMatcher) {
+      ((TargetMatcher) parent).setMatcher((SymbolProblemTargetMatcher) child);
+    } else {
+      throw new SAXException(
+          "Unexpected parent-child relationship. Parent:" + parent + ", child:" + child);
+    }
+  }
+
+  @Override
+  public void startElement(
+      String namespaceURI, String localName, String qName, Attributes attributes)
+      throws SAXException {
+    if (!namespaceURI.isEmpty()) {
+      throw new SAXException("The exclusion rule does not support XML namespace");
+    }
+    switch (localName) {
+      case "LinkageCheckerFilter":
+        break;
+      case "LinkageError":
+        stack.push(new LinkageErrorMatcher());
+        break;
+      case "Source":
+        SourceMatcher sourceMatcher = new SourceMatcher();
+        ((LinkageErrorMatcher) stack.peek()).setSourceMatcher(sourceMatcher);
+        stack.push(sourceMatcher);
+        break;
+      case "Target":
+        TargetMatcher targetMatcher = new TargetMatcher();
+        ((LinkageErrorMatcher) stack.peek()).setTargetMatcher(targetMatcher);
+        stack.push(targetMatcher);
+        break;
+      case "Package":
+        PackageMatcher packageMatcher = new PackageMatcher(attributes.getValue("name"));
+        addMatcherToTop(packageMatcher);
+        break;
+      case "Class":
+        String classNameOnClass = attributes.getValue("name");
+        ClassMatcher classMatcher = new ClassMatcher(classNameOnClass);
+        addMatcherToTop(classMatcher);
+        break;
+      case "Method":
+        String classNameOnMethod = attributes.getValue("className");
+        MethodMatcher methodMatcher =
+            new MethodMatcher(classNameOnMethod, attributes.getValue("name"));
+        addMatcherToTop(methodMatcher);
+        break;
+      case "Field":
+        String classNameOnField = attributes.getValue("className");
+        FieldMatcher fieldMatcher = new FieldMatcher(classNameOnField, attributes.getValue("name"));
+        addMatcherToTop(fieldMatcher);
+        break;
+      default:
+        throw new SAXException("Unknown tag " + localName);
```

Comment:
It's OK and arguably better to simply ignore unrecongized elements.

Reply:
I prefer to show errors for unrecognized elements. The errors help to detect typos.

More validation will be done in https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1277